src/main/java/de/tum/in/www1/artemis/web/rest/programming/ProgrammingExerciseResource.java

```diff
@@ -303,9 +303,14 @@ public ResponseEntity<ProgrammingExercise> updateProgrammingExercise(@RequestBod
                 updatedProgrammingExercise.getBuildConfig().isTestwiseCoverageEnabled())) {
             throw new BadRequestAlertException("Testwise coverage enabled flag must not be changed", ENTITY_NAME, "testwiseCoverageCannotChange");
         }
-        if (!Boolean.TRUE.equals(updatedProgrammingExercise.isAllowOnlineEditor()) && !Boolean.TRUE.equals(updatedProgrammingExercise.isAllowOfflineIde())) {
-            return ResponseEntity.badRequest().headers(HeaderUtil.createAlert(applicationName,
-                    "You need to allow at least one participation mode, the online editor or the offline IDE", "noParticipationModeAllowed")).body(null);
+        if (!Boolean.TRUE.equals(updatedProgrammingExercise.isAllowOnlineEditor()) && !Boolean.TRUE.equals(updatedProgrammingExercise.isAllowOfflineIde())
+                && !updatedProgrammingExercise.isAllowOnlineIde()) {
+            throw new BadRequestAlertException("You need to allow at least one participation mode, the online editor, the offline IDE, or the online IDE", ENTITY_NAME,
+                    "noParticipationModeAllowed");
+        }
+        // Verify that a theia image is provided when the online IDE is enabled
+        if (updatedProgrammingExercise.isAllowOnlineIde() && updatedProgrammingExercise.getBuildConfig().getTheiaImage() == null) {
+            throw new BadRequestAlertException("You need to provide a Theia image when the online IDE is enabled", ENTITY_NAME, "noTheiaImageProvided");
```

Comment:
Should this also verify that the `theia` profile is active on the server if online IDEs are allowed?